Holders patch

DalwynWasTaken opened the pull request:
Fix the issues regarding diagnosed crashes

Added comments to unclear sections and made a variable be more descriptive to what it contains

DalwynWasTaken committed:
Holders patch

Fix the issues regarding diagnosed crashes

Added comments to unclear sections and made a variable be more descriptive to what it contains

kristofbolyai commented on common/src/main/java/com/wynntils/core/config/objects/ConfigHolder.java:
```diff
@@ -9,14 +9,19 @@
+                // This is a check for if the feature doesn't have the FeatureInfo annotation
+                parent.getClass().getAnnotation(FeatureInfo.class) != null
+                        ? parent.getClass().getAnnotation(FeatureInfo.class).category()
+                        : "",
```
Null checking seems not to be required. See #109, which seems to work fine.

kristofbolyai replied:
Yeah, no need for the null checking, otherwise we can merge this.

kristofbolyai commented on common/src/main/java/com/wynntils/core/features/FeatureRegistry.java:
```diff
@@ -100,8 +100,8 @@ private static void initializeFeature(Feature feature) {
-        for (Field f : FieldUtils.getFieldsWithAnnotation(feature.getClass(), Config.class)) {
-            ConfigManager.registerHolder(new ConfigHolder(feature.getClass(), f, f.getAnnotation(Config.class)));
+        for (Field field : FieldUtils.getFieldsWithAnnotation(feature.getClass(), Config.class)) {
+            ConfigManager.registerHolder(new ConfigHolder(feature, field, field.getAnnotation(Config.class)));
```
Also can you revert to the original field name so we don't have to merge other PRs?

DalwynWasTaken committed:
Tweak

Remove unneeded null check

DalwynWasTaken committed:
Tweak #2

Revert variable naming to prevent the need of merging all other development branches
